GoogleForCreators/web-stories-wp
# Added remaining layer lock handling

## Context

This adds a bunch of extra features related to locked layers. Hopefully everything to make the feature fully functional.

See todo-list for details.

## Technical choices
Some simplifications and assumptions for technical reasons:
* Multi-selection is not possible if any element is locked. This makes multi-selection transformations easier to handle (they're always allowed because no element in selection can be locked)
* You cannot right-click a locked element directly, but once selected, you can. You can however no longer "click through" a locked element once selected.

Both of the above scenarios deviate from Figma's handling of locked elements, but both of the above solutions were much easier to implement than to copy how Figma does them.

## To-do

* [x] When dragging media, do not highlight locked media and shape as potential drop target nor allow media to be dropped "into" locked elements
* [x] When hovering a locked element, do not display a slight blue border
* [x] Do not allow locked elements to be clicked, unless already selected (can be selected through right-click menu or layer panel)
* [x] Filter out locked elements from lasso selection
* [x] Filter out locked elements from shift-selection in layer panel
* [x] When meta-selecting a non-locked layer via layer panel, if old selection was a locked layer, only select new element
* [x] When meta-selecting a locked layer via layer panel, forget old selection and only select the locked layer
* [x] When shift-selecting a non-locked layer via canvas, if old selection was a locked layer, only select new element
* [x] Do not show links when hovering locked elements
* [x] When a locked element is selected, do not allow any transformation of the element using moveable (hide all handles and use default cursors)
* [x] When a locked media or shape element is selected, allow right-click to open context menu
* [x] When a locked non-media/non-shape element is selected, allow right-click to open context menu
* [x] When a locked element is selected, do not allow movement via keyboard arrows
* [x] When focusing a locked element on canvas, make lock status part of element accessible name
* [x] When editing a locked media element, do not allow resizing the frame
* [x] When focusing a locked element, announce lock state as part of element name for screen-readers.
* [x] When a locked text element is selected, do not allow edit mode using enter key, any key, or click inside
* [x] Add unit tests for changes to story reducer
* [x] Add karma tests for changes to layer selection
* [x] Add karma tests for changes to lasso selection
* [x] Add karma tests for changes to canvas selection

## User-facing changes

_TODO_

## Testing Instructions

This PR can be tested by following these steps:

1. Add multiple elements to canvas of different types
2. Stack some on top of each other
3. Lock some layers, especially some of the top ones
4. Observe that layer and element selection makes sense both using canvas, lasso, and layer panel.

## Checklist




---

Fixes #11428
